cfpb/consumerfinance.gov
# WIP: RAF tool i18n and configurable county threshold

Adds a language data attribute to the container element in the RAF tool template and updates the RAF JS to include translations that enable a Spanish-language version of the tool. (Note: the JS should be imported from an npm module when PR is ready for final review)

## Additions

- `data-language` attribute on container element in RAF tool template to pass the page's language to the tool
- New RAF tool struct block with optional `county-threshold` field
- `data-county-threshold` attribute on container element in RAF tool template to pass the optional `county-threshold` value to the tool

## Changes

- Update RAF JS

## How to test this PR

1. Check out branch and run gulp copy & migrations
2. Replace existing `Rental Assistance Finder Tool` block on [English RAF tool page](http://localhost:8000/coronavirus/mortgage-and-housing-assistance/renter-protections/find-help-with-rent-and-utilities/) with new `RAF Tool (configurable)` block 
3. Navigate to [English RAF tool page](http://localhost:8000/coronavirus/mortgage-and-housing-assistance/renter-protections/find-help-with-rent-and-utilities/) and verify that English-language tool content displays as expected
4. In admin, create a new browse page and set page language to Spanish. Add `RAF tool (configurable)` and verify that tool content displays in Spanish.
5. Change language of page to anything other than Spanish or English and verify that tool displays in English
6. Verify that by default, tool shows county dropdown when a state with more than 10 results is selected rather than old threshold of 5 (New Jersey should show county dropdown, Arizona should not)
7. Change `county_threshold` value in RAF Tool block and verify that tool uses new threshold value instead of default 10

## Screenshots

<img width="879" alt="Screen Shot 2021-08-12 at 1 40 26 PM" src="https://user-images.githubusercontent.com/778171/129397215-f198fb14-37db-4b22-b056-89288162d20e.png">

## Notes and todos

- This PR includes copied RAFT JS. We will either update this PR or open a new one immediately afterwards that imports the JS from a recently-published npm module.

## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
